requirements.txt

```diff
@@ -1,14 +1,14 @@
 absl-py
 packaging
 pandas
-tensorflow
+tf-nightly==2.16.0.dev20231026  # Pin a working nightly until rc0.
```

Comment:
With this change are we breaking compatibility with older TF?  Also, TF 2.16 is not going to be released until Feb'24.  Can we be compatible with TF 2.15 which will work with Keras 3 except for pip compatibility check?

Reply:
Also the GitHub actions all use TF 2.13, so I think we need to think through the min TF Version we want to be compatible with.

Reply:
@sampathweb I am trying to mirror keras_nlp - https://github.com/keras-team/keras-nlp/pull/1273/files

Reply:
If we are doing this, we need to update GitHub actions as well and also agree that we are dropping support for TF < 2.16.  I don't think we are ready to do that yet.  So wanted to discuss this change further.

Reply:
As per our discussion, I have changed the requirements to use stable version of TF. This PR will only change the names of the tests.

Reply:
Oops sorry probably miscommunication here! I had thought we had discussed having a test run that test against a stable version of tf, but *also* have multiple tests run that checks Keras 3 on all backends. Let's hit both the compat check and the new nightly check in our testing setup.

Here's how we are doing this on KerasNLP:
https://github.com/keras-team/keras-nlp/blob/master/.github/workflows/actions.yml#L30-L34
and
https://github.com/keras-team/keras-nlp/blob/master/.github/workflows/actions.yml#L66-L69

If we want to test keras-cv against an earlier version of tensorflow like 2.13 we can do that by updating the version pin in the former link. I don't think we can spare the compute to test every tf version, but this way we will span checking the oldest version of tf we would like to support, and the newest (tf and keras 2.16).

(also note the keras-nlp has a slightly different setup for requirements files. it would be great to consolidate at some point, but probably a separate pr/discussion)

Reply:
@sampathweb @divyashreepathihalli does that make sense to both of you? If not we can sync up!

Reply:
Agree.  We should keep similar setup for `keras-cv` where we test stable version of TensorFlow 2.13 or 2.14 with Keras 2 setup and use pinned `tf-nightly` with Keras 3 for multibackend similar to how its done in `keras-nlp`.

Reply:
updated requuirements and actions.file to tests againts tf 2.14 for keras 2 and nightly version of tf for keras 3